fdbcli/DebugCommands.actor.cpp

```diff
@@ -289,35 +262,15 @@ std::tuple<bool, bool, Key> checkResults(Version version,
 			}
 		}
 	}
-	return std::make_tuple(allSame, hasMore, claimEndKey);
+	return allSame;
 }
 
-// The command is used to check the inconsistency in a keyspace, default is \xff\x02/blog/ keyspace.
-ACTOR Future<bool> checkallCommandActor(Database cx, std::vector<StringRef> tokens) {
-	state Transaction onErrorTr(cx); // This transaction exists only to access onError and its backoff behavior
-	state bool checkAll = false; // If set, do not return on first error, continue checking all keys
-	state KeyRange inputRange;
-	if (tokens.size() == 3) {
-		inputRange = KeyRangeRef(tokens[1], tokens[2]);
-	} else if (tokens.size() == 4 && tokens[3] == "all"_sr) {
-		inputRange = KeyRangeRef(tokens[1], tokens[2]);
-		checkAll = true;
-	} else {
-		printf(
-		    "checkall [<KEY> <KEY2>] (all)\n"
-		    "Check inconsistency of the input range by comparing all replicas and print any corruptions.\n"
-		    "The default behavior is to stop on the first subrange where corruption is found\n"
-		    "`all` is optional. When `all` is appended, the checker does not stop until all subranges have checked.\n"
-		    "Note this is intended to check a small range of keys, not the entire database (consider consistencycheck "
-		    "for that purpose).\n");
-		return false;
-	}
-	if (inputRange.empty()) {
-		printf("Input empty range: %s.\nImmediately exit.\n", printable(inputRange).c_str());
-		return true;
-	}
+ACTOR Future<bool> doCheckAll(Database cx, KeyRange inputRange, bool checkAll);
 
-	// At this point, we have a non-empty inputRange to check
+// Return whether inconsistency is detected in the inputRange
+ACTOR Future<bool> doCheckAll(Database cx, KeyRange inputRange, bool checkAll) {
+	state Transaction onErrorTr(cx); // This transaction exists only to access onError and its backoff behavior
+	state bool consistent = true;
```

Comment:
it looks like we do not need both `allSame` and `consistent`, only var is needed

Reply:
only 1 var is needed*

Reply:
consistent is the doCheckAll() wide result. allSame is only for checkResult of the current round.